File: public/docs-assets/examples/languages/quran-all-in-one-request.py
# Python Example: All-in-One Request
# This example shows how to fetch multiple data types in a single request using the GlobalQuran API

import requests
import json

# API Endpoint
endpoint = 'https://api.globalquran.com/v1/all/surah/1/quran-simple/en?key=YOUR_API_KEY'

def fetch_all_data(data_in='surah', data_in_no=1, quran_id='quran-simple', lang_code='en'):
    """
    Function to fetch multiple data types in a single request
    
    Args:
        data_in (str): Type of data to fetch (surah, ayah, page, juz)
        data_in_no (int): Number for the data type
        quran_id (str): Quran ID (default: 'quran-simple')
        lang_code (str): Language code (default: 'en')
    
    Returns:
        dict: API response data or None if error
    """
    try:
        url = f"https://api.globalquran.com/v1/all/{data_in}/{data_in_no}/{quran_id}/{lang_code}?key=YOUR_API_KEY"
        response = requests.get(url)
        response.raise_for_status()  # Raises an HTTPError for bad responses
        
        data = response.json()
        return data
        
    except requests.exceptions.RequestException as e:
        print(f"Error fetching all data: {e}")
        return None
    except json.JSONDecodeError as e:
        print(f"Error parsing JSON: {e}")
        return None

def display_all_data(data):
    """
    Display all data in a formatted way
    
    Args:
        data (dict): API response data
    """
    if not data:
        print("No data available")
        return
    
    print("All-in-One API Response:")
    print("=" * 50)
    
    # Display Quran data
    if 'quran' in data:
        print("Quran Data:")
        quran_data = data['quran']
        for surah_key, surah_data in quran_data.items():
            print(f"  Surah {surah_key}: {len(surah_data)} ayahs")
            # Show first few ayahs
            for ayah_key, ayah_data in list(surah_data.items())[:3]:
                print(f"    Ayah {ayah_key}: {ayah_data.get('verse', 'N/A')[:50]}...")
        print()
    
    # Display translation data
    if 'translation' in data:
        print("Translation Data:")
        translation_data = data['translation']
        for surah_key, surah_data in translation_data.items():
            print(f"  Surah {surah_key}: {len(surah_data)} translations")
            # Show first few translations
            for ayah_key, ayah_data in list(surah_data.items())[:3]:
                print(f"    Ayah {ayah_key}: {ayah_data.get('text', 'N/A')[:50]}...")
        print()
    
    # Display recitor data
    if 'recitor' in data:
        print("Recitor Data:")
        recitor_data = data['recitor']
        for surah_key, surah_data in recitor_data.items():
            print(f"  Surah {surah_key}: {len(surah_data)} recitations")
        print()

# Main execution
if __name__ == "__main__":
    print("Fetching all-in-one data for Surah 1...")
    data = fetch_all_data('surah', 1, 'quran-simple', 'en')
    
    if data:
        display_all_data(data)
        
        print("Example usage for different data types:")
        print("Surah: https://api.globalquran.com/v1/all/surah/1/quran-simple/en?key=YOUR_API_KEY")
        print("Ayah: https://api.globalquran.com/v1/all/ayah/2:255/quran-simple/en?key=YOUR_API_KEY")
        print("Page: https://api.globalquran.com/v1/all/page/1/quran-simple/en?key=YOUR_API_KEY")
        print("Juz: https://api.globalquran.com/v1/all/juz/1/quran-simple/en?key=YOUR_API_KEY")
    else:
        print("Failed to fetch all-in-one data")

# Expected Response Structure:
# {
#   "quran": {
#     "1": {
#       "1": { "surah": 1, "ayah": 1, "verse": "..." },
#       "2": { "surah": 1, "ayah": 2, "verse": "..." }
#     }
#   },
#   "translation": {
#     "1": {
#       "1": { "surah": 1, "ayah": 1, "text": "..." },
#       "2": { "surah": 1, "ayah": 2, "text": "..." }
#     }
#   },
#   "recitor": {
#     "1": {
#       "1": { "surah": 1, "ayah": 1, "audio": "..." },
#       "2": { "surah": 1, "ayah": 2, "audio": "..." }
#     }
#   }
# }

File: public/docs-assets/examples/languages/quran-all-in-one-request.js
// JavaScript Example: All-in-One Request
// This example shows how to fetch multiple data types in a single request using the GlobalQuran API

// API Endpoint
const endpoint = 'https://api.globalquran.com/v1/all/surah/1/quran-simple/en?key=YOUR_API_KEY';

// Function to fetch all-in-one data
async function fetchAllInOneData(dataIn = 'surah', dataInNo = 1, quranId = 'quran-simple', langCode = 'en') {
    try {
        const url = `https://api.globalquran.com/v1/all/${dataIn}/${dataInNo}/${quranId}/${langCode}?key=YOUR_API_KEY`;
        const response = await fetch(url);
        
        if (!response.ok) {
            throw new Error(`HTTP error! status: ${response.status}`);
        }
        
        const data = await response.json();
        return data;
    } catch (error) {
        console.error('Error fetching all-in-one data:', error);
        return null;
    }
}

// Function to display all data
function displayAllData(data) {
    if (!data) {
        console.log('No data available');
        return;
    }
    
    console.log('All-in-One API Response:');
    console.log('========================');
    
    // Display Quran data
    if (data.quran) {
        console.log('Quran Data:');
        Object.keys(data.quran).forEach(surahKey => {
            const surah = data.quran[surahKey];
            const ayahCount = Object.keys(surah).length;
            console.log(`  Surah ${surahKey}: ${ayahCount} ayahs`);
            
            // Show first few ayahs
            Object.keys(surah).slice(0, 3).forEach(ayahKey => {
                const ayah = surah[ayahKey];
                console.log(`    Ayah ${ayahKey}: ${ayah.verse.substring(0, 50)}...`);
            });
        });
        console.log();
    }
    
    // Display translation data
    if (data.translation) {
        console.log('Translation Data:');
        Object.keys(data.translation).forEach(surahKey => {
            const surah = data.translation[surahKey];
            const ayahCount = Object.keys(surah).length;
            console.log(`  Surah ${surahKey}: ${ayahCount} translations`);
            
            // Show first few translations
            Object.keys(surah).slice(0, 3).forEach(ayahKey => {
                const ayah = surah[ayahKey];
                console.log(`    Ayah ${ayahKey}: ${ayah.text.substring(0, 50)}...`);
            });
        });
        console.log();
    }
    
    // Display recitor data
    if (data.recitor) {
        console.log('Recitor Data:');
        Object.keys(data.recitor).forEach(surahKey => {
            const surah = data.recitor[surahKey];
            const ayahCount = Object.keys(surah).length;
            console.log(`  Surah ${surahKey}: ${ayahCount} recitations`);
        });
        console.log();
    }
}

// Function to get specific ayah from all data
function getAyahFromAllData(data, surahNumber, ayahNumber) {
    if (!data) return null;
    
    const result = {};
    
    if (data.quran && data.quran[surahNumber] && data.quran[surahNumber][ayahNumber]) {
        result.quran = data.quran[surahNumber][ayahNumber];
    }
    
    if (data.translation && data.translation[surahNumber] && data.translation[surahNumber][ayahNumber]) {
        result.translation = data.translation[surahNumber][ayahNumber];
    }
    
    if (data.recitor && data.recitor[surahNumber] && data.recitor[surahNumber][ayahNumber]) {
        result.recitor = data.recitor[surahNumber][ayahNumber];
    }
    
    return Object.keys(result).length > 0 ? result : null;
}

// Function to count total ayahs
function countTotalAyahs(data) {
    if (!data || !data.quran) return 0;
    
    let total = 0;
    Object.keys(data.quran).forEach(surahKey => {
        total += Object.keys(data.quran[surahKey]).length;
    });
    
    return total;
}

// Main execution
async function main() {
    console.log('Fetching all-in-one data for Surah 1...');
    const data = await fetchAllInOneData('surah', 1, 'quran-simple', 'en');
    
    if (data) {
        displayAllData(data);
        
        const totalAyahs = countTotalAyahs(data);
        console.log(`Total ayahs in response: ${totalAyahs}`);
        
        // Example: Get specific ayah
        const ayah = getAyahFromAllData(data, 1, 1);
        if (ayah) {
            console.log('\nSpecific Ayah (1:1):');
            if (ayah.quran) {
                console.log(`Quran: ${ayah.quran.verse}`);
            }
            if (ayah.translation) {
                console.log(`Translation: ${ayah.translation.text}`);
            }
        }
        
        console.log('\nExample usage:');
        console.log('const data = await fetchAllInOneData("surah", 1, "quran-simple", "en");');
        console.log('const ayah = getAyahFromAllData(data, 1, 1);');
        console.log('const count = countTotalAyahs(data);');
    } else {
        console.log('Failed to fetch all-in-one data');
    }
}

// Run the example
main();

// Expected Response Structure:
// {
//   "quran": {
//     "1": {
//       "1": { "surah": 1, "ayah": 1, "verse": "..." },
//       "2": { "surah": 1, "ayah": 2, "verse": "..." }
//     }
//   },
//   "translation": {
//     "1": {
//       "1": { "surah": 1, "ayah": 1, "text": "..." },
//       "2": { "surah": 1, "ayah": 2, "text": "..." }
//     }
//   },
//   "recitor": {
//     "1": {
//       "1": { "surah": 1, "ayah": 1, "audio": "..." },
//       "2": { "surah": 1, "ayah": 2, "audio": "..." }
//     }
//   }
// }
